### PR TITLE
Minifier fixes

### PR DESCRIPTION
**Pull request by anijain2305**
* bring back repro level 3
* same ux for repro_after="dynamo" and "aot"
* fake tensor fix refactor

### CODE REVIEW

File: torchdynamo/debug_utils.py [hunk 1]
```diff
@@ -427,114 +445,108 @@ def generate_dynamo_fx_repro_string(model_str, args, compiler_name):
...
-    shutil.copytree(tmp_dir, local_dir)
-    local_tar_file = os.path.join(torchdynamo.config.base_dir, "repro.tar.gz")
-    print(f"Writing checkpoint with {len(gm.graph.nodes)} locally to {local_tar_file}")
-    with tarfile.open(local_tar_file, "w:gz") as tar:
-        tar.add(local_dir, arcname=os.path.basename(local_dir))
+    except OSError:
+        log.warning("No write permissions for {local_path}")
+
+
+# def dump_backend_repro_as_tarfile(gm, args, compiler_name):
```
Comment (Chillee) [inline torchdynamo/debug_utils.py:500]: leftover?

**Review by Chillee — APPROVED** [review]
API LGTM, did not review refactor of minifier extensively.

File: torchdynamo/debug_utils.py (hunk 1 above)
Comment (anijain2305) [reply torchdynamo/debug_utils.py:500]: I want to keep it if we ever need it back again.

Comment (Chillee) [reply torchdynamo/debug_utils.py:500]: Probably keep it uncommented but add a note.

Comment (anijain2305) [reply torchdynamo/debug_utils.py:500]: Ok will do.

File: torchdynamo/debug_utils.py [hunk 2]
```diff
@@ -323,39 +341,38 @@ def debug_wrapper(gm, example_inputs, **kwargs):
         orig_graph = copy.deepcopy(gm.graph)
         assert config.repro_after in ("dynamo", "aot", None)
 
-        def wrap_fn_to_repro(fn, example_inputs):
+        def deferred_for_real_inputs(*real_inputs):
```
[the remark this answers is not in the document]
Comment (eellison) [reply torchdynamo/debug_utils.py:344]: looks good

**Review by eellison — APPROVED** [review]

**Pull request merged**